wfe/web-front-end_test.go

```diff
@@ -1021,3 +1034,62 @@ func TestTermsRedirect(t *testing.T) {
 		agreementURL)
 	test.AssertEquals(t, responseWriter.Code, 302)
 }
+
+func TestIssuer(t *testing.T) {
+	wfe := setupWFE(t)
+	wfe.IssuerCacheDuration = time.Second * 10
+	wfe.IssuerCert = []byte{0, 0, 1}
+
+	responseWriter := httptest.NewRecorder()
+
+	wfe.Issuer(responseWriter, &http.Request{
+		Method: "GET",
+	})
+	test.AssertEquals(t, responseWriter.Code, http.StatusOK)
+	test.Assert(t, bytes.Compare(responseWriter.Body.Bytes(), wfe.IssuerCert) == 0, "Incorrect bytes returned")
+	test.AssertEquals(t, responseWriter.Header().Get("Cache-Control"), "public, max-age=10")
+}
+
+func TestGetCertificate(t *testing.T) {
+	wfe := setupWFE(t)
+	wfe.CertCacheDuration = time.Second * 10
+	wfe.CertNoCacheExpirationWindow = time.Hour * 24 * 7
+	wfe.SA = &MockSA{}
+
+	certPemBytes, _ := ioutil.ReadFile("test/178.crt")
+	certBlock, _ := pem.Decode(certPemBytes)
+
+	responseWriter := httptest.NewRecorder()
+
+	path, _ := url.Parse("/acme/cert/00000000000000b2")
+	wfe.Certificate(responseWriter, &http.Request{
+		Method: "GET",
+		URL:    path,
+	})
+	test.AssertEquals(t, responseWriter.Code, 200)
+	test.AssertEquals(t, responseWriter.Header().Get("Cache-Control"), "public, max-age=10")
+	test.AssertEquals(t, responseWriter.Header().Get("Content-Type"), "application/pkix-cert")
+	test.Assert(t, bytes.Compare(responseWriter.Body.Bytes(), certBlock.Bytes) == 0, "Certificates don't match")
+
+	responseWriter = httptest.NewRecorder()
+	path, _ = url.Parse("/acme/cert/00000000000000ff")
+	wfe.Certificate(responseWriter, &http.Request{
+		Method: "GET",
+		URL:    path,
+	})
+	test.AssertEquals(t, responseWriter.Code, 404)
+	test.AssertEquals(t, responseWriter.Header().Get("Cache-Control"), "public, max-age=0, no-cache")
+	test.AssertEquals(t, responseWriter.Body.String(), `{"type":"urn:acme:error:malformed","detail":"Not found"}`)
+
+	responseWriter = httptest.NewRecorder()
+	path, _ = url.Parse("/acme/cert/0000000000000001")
```

Comment:
I'll add some comments to indicate what each of these are testing and add a bit more flesh to the `wfe.Certificate` tests that are currently lacking.